Dockerfile

```diff
@@ -113,66 +145,72 @@ RUN apt-get update && apt-get install -y nodejs build-essential npm && \
     hash -d npm \
     npm install -g xml-js
 
-# Add moustache templates as mo
-RUN curl -sSO https://raw.githubusercontent.com/tests-always-included/mo/master/mo && \
-    chmod +x mo && \
-    mv mo /usr/local/bin
-
 # Clean up package manager
 RUN apt-get clean autoclean
 
 # Add random number generator to skip Docker building cache
 ADD http://www.random.org/strings/?num=10&len=8&digits=on&upperalpha=on&loweralpha=on&unique=on&format=plain&rnd=new /uuid
 
-# Elpis
+
+########################## ELPIS INSTALLATION ########################
+
 WORKDIR /
+
+# Elpis
 RUN git clone --depth=1 https://github.com/CoEDL/elpis.git
 
 # Elpis GUI
 RUN git clone --depth=1 https://github.com/CoEDL/elpis-gui.git
 
-# Example data
+WORKDIR /elpis-gui
+
+RUN npm install && \
+    npm run build
+
 WORKDIR /tmp
+
+# Example data
 RUN git clone --depth=1 https://github.com/CoEDL/toy-corpora.git
 
-# Set up zsh & ENV variables
+
+########################## ZSH CONFIG & ENV  ##########################
+
 WORKDIR /root
+
+# ZSH config
 RUN apt-get install zsh
 RUN chsh -s /usr/bin/zsh root
 RUN sh -c "$(wget -O- https://raw.githubusercontent.com/deluan/zsh-in-docker/master/zsh-in-docker.sh)" -- -t robbyrussell -p history-substring-search -p git
+
+# ENV
 RUN echo "export FLASK_ENV=development" >> ~/.zshrc
 RUN echo "export FLASK_APP=elpis" >> ~/.zshrc
 RUN echo "export LC_ALL=C.UTF-8" >> ~/.zshrc
 RUN echo "export LANG=C.UTF-8" >> ~/.zshhrc
 RUN echo "export PATH=$PATH:/venv/bin:/kaldi/src/bin/" >> ~/.zshrc
 RUN echo "alias run=\"flask run --host=0.0.0.0 --port=5000\"" >> ~/.zshrc
+RUN echo "alias run-win=\"flask run --host=127.0.0.1 --port=5000\"" >> ~/.zshrc
 RUN cat ~/.zshrc >> ~/.bashrc
-
-# Move ENV lines up. Putting here for now to build on top of cached builds
+# Make ENV vars available for non-interactive running
 ENV FLASK_ENV='development'
 ENV FLASK_APP='elpis'
 ENV LC_ALL=C.UTF-8
 ENV LANG=C.UTF-8
+# Don't do this, it breaks `python setup.py develop` below
+#RUN source $HOME/.bashrc
 
-WORKDIR /elpis-gui
-RUN npm install && \
-    npm run build
+
+########################## RUN THE APP ##########################
 
 WORKDIR /elpis
+
 RUN /usr/bin/python3 -m venv /venv
+
 ENV PATH="/venv/bin:$PATH"
 
-RUN pip3.6 install wheel pytest pylint && python setup.py develop
+RUN apt-get update && apt-get install libffi-dev
```

Comment:
It shouldn't be necessary to call update again here. Any chance this could be added to the previous install step(s)?

Reply:
Will try it again without, but seemed that occasionally installation failed (here and earlier stages) that was fixed fix updating.

Reply:
Looks like that whole line can go.